rubygems-plugin/rubygems_plugin.rb

```diff
@@ -1,11 +1,18 @@
-# Yes borrowed from rbenv. Couldn't take my mind off that implementation
-
-Gem.post_install do |installer|
-
-  if installer.spec.executables.any? && installer.bin_dir == Gem.default_bindir
-    installer.spec.executables.each do |executable|
-     `asdf reshim ruby #{RUBY_VERSION} bin/#{executable}`
-    end
+module ReshimInstaller
+  def install(options)
+    super
+    # We don't know which gems were installed, so always reshim.
+    `asdf reshim ruby`
   end
+end
 
+if defined?(Bundler::Installer)
+  Bundler::Installer.prepend ReshimInstaller
+else
+  maybe_reshim = lambda do |installer|
+    # If any gems with executables were installed or uninstalled, reshim.
+    `asdf reshim ruby` if installer.spec.executables.any?
```

Comment:
Why not specify the `RUBY_VERSION` here?

Reply:
I understand why the `bin/${executable}` part was removed, but the version might still be worth including.

Reply:
I'm not sure if this is a bug with asdf itself, but while the `#{RUBY_VERSION}` does seem to work after `gem install`, it does not remove the dead shim after `gem uninstall`.

With version:
```
$ gem install railties ; echo "Rails: $(which rails)" ; yes | gem uninstall --force railties; echo "Rails: $(which rails)"
Fetching railties-6.0.2.2.gem
Successfully installed railties-6.0.2.2
Parsing documentation for railties-6.0.2.2
Installing ri documentation for railties-6.0.2.2
Done installing documentation for railties after 0 seconds
1 gem installed
Rails: /Users/djmarcin/.asdf/shims/rails
Removing rails
Successfully uninstalled railties-6.0.2.2
Rails: /Users/djmarcin/.asdf/shims/rails
```

Without version:
```
$ gem install railties ; echo "Rails: $(which rails)" ; yes | gem uninstall --force railties; echo "Rails: $(which rails)"
Fetching railties-6.0.2.2.gem
Successfully installed railties-6.0.2.2
Parsing documentation for railties-6.0.2.2
Installing ri documentation for railties-6.0.2.2
Done installing documentation for railties after 0 seconds
1 gem installed
Rails: /Users/djmarcin/.asdf/shims/rails
Removing rails
Successfully uninstalled railties-6.0.2.2
Rails:
```

Although, since installs are likely more frequent than uninstalls, I could change the behavior to iterate over the binaries on install, and just reshim everything on uninstall.

Reply:
I've added the customized behavior for install & uninstall.